### DISCUSSION TITLE
Best practice to count group members

### DISCUSSION THREAD

**Discussion by klepptor** [DirectoryTree/LdapRecord#512] [Q&A] [answered]
Hi there!

I need to count the members of several Active Directory groups. This is my code:

````
            $group = Group::findByOrFail('distinguishedName', $groupDN);
            $members = $group->members()->get();
            $count_members = sizeof($members);
````

This works, but it's quite slow with >10 groups and  >1500 members in each group.

Is there a better (faster) way to achieve my goal?

Thx!

**Comment by stevebauman** [comment, marked as answer]
Hi @klepptor!

I've just added a method for this in the latest [v2.18.0](https://github.com/DirectoryTree/LdapRecord/releases/tag/v2.18.0) release.

This method (`count()`) will execute a query on your LDAP server for all members the group, but restrict the query to only return `objectclass`, trimming down the result size and speeding up the result.

Here is how you can utilize this method:

```php
$group = Group::findByOrFail('distinguishedName', $groupDN);

$countOfMembers = $group->members()->count();
```

However, the fastest way would be to count the `member` attribute on your resulting `$group` model, if your server returns the full list (which may not be the case with larger groups):

```php
// Notice the last argument in the below method, asking the LDAP
// server to only return the "member" attribute, speeding
// up the query due to smaller response size:
$group = Group::findByOrFail('distinguishedName', $groupDN, 'member');

$countOfMembers = count($group['member']);
```

**Reply by klepptor** [reply]
Great! @stevebauman 

Thx a lot (and again) for your work!!

**Reply by stevebauman** [reply]
Happy to help @klepptor! 😄
